### ISSUE TITLE
[bug] Cannot update certificate thumbprints

### ISSUE THREAD

**Issue by nanowireUK** [Azure/azure-iot-cli-extension#296]
**Describe the bug**
Trying to do a `az iot hub device-identity update` for updating either primary and secondary thumbprints fails to make the desired changes but reports no error. The returned response still shows the original thumbprints. Example commands tried:
```bash
az iot hub device-identity update -d l5edge -n my-iothub --ptp 781DE5AEEECCC7FE857FF35C393FDED1DF39595D
az iot hub device-identity update -d l5edge -n my-iothub --ptp "781DE5AEEECCC7FE857FF35C393FDED1DF39595D"
az iot hub device-identity update -d l5edge -n my-iothub --stp 781DE5AEEECCC7FE857FF35C393FDED1DF39595D
az iot hub device-identity update -d l5edge -n my-iothub --stp "781DE5AEEECCC7FE857FF35C393FDED1DF39595D"
```

Work around has been to use the `set` command instead:
`az iot hub device-identity update -d l5edge -n my-iothub --set authentication.x509Thumbprint.primaryThumbprint=781DE5AEEECCC7FE857FF35C393FDED1DF39595D`

**To Reproduce**

1. Install az-cli and azure-iot extension
2. Create IoT Edge device in portal with self-signed certficate thumbprints
3. After az login try command `az iot hub device-identity update -d l5edge -n my-iothub --ptp 781DE5AEEECCC7FE857FF35C393FDED1DF39595D`

**Expected behavior**
The thumbprint should be updated.

**Screenshots**

**Environment (please complete the following information):**

- OS: Ubuntu 20.04 on WSL
- Shell: wsl bash
- Az CLI version: 2.18.0
- IoT extension version: 0.10.8
- Python version (if pip installed): Python 3.8.5

**Additional context**

**Comment by nanowireUK** [comment]
Looking at test code, this isn't being picked up because the test doesn't change the primary or secondary thumbprint. It only updates it to the original value which would pass fine since there are no errors.

**Comment by c-ryan-k** [comment]
@nanowireUK  - thanks for making us aware of this - taking a look now

**Comment by c-ryan-k** [comment]
@nanowireUK - 
What's happening here is that our update command is only checking those thumbprints (or keys for `sas` auth) if you have provided an `auth_method` to the update command (which covers the case of switching `auth_method`, but not updating it ;)

Your `--set` workaround should work fine, as should adding `--am x509_thumbprint` to the update command while I add a clause to fix this.

**Comment by digimaun** [comment]
This should be fixed in our recently released `0.10.9`. Please update and give it a try. Closing this issue, feel free to comment here and we can re-open.
